### PR TITLE
DATAES-415 - More Field Types Support

### PR DESCRIPTION
**Pull request by zzt93**
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAES).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

File: src/main/java/org/springframework/data/elasticsearch/annotations/FieldType.java [hunk 1]
```diff
@@ -19,12 +19,17 @@
  * @author Rizwan Idrees
  * @author Mohsin Husen
  * @author Artur Konczak
+ * @author zzt
```
[the remark this answers is not in the document]
Comment (zzt93) [reply src/main/java/org/springframework/data/elasticsearch/annotations/FieldType.java]: Fixed

File: src/main/java/org/springframework/data/elasticsearch/annotations/FieldType.java [hunk 2]
```diff
@@ -19,12 +19,17 @@
...
  */
 public enum FieldType {
 	Text,
+  Byte, 
+  Short, 
 	Integer,
 	Long,
 	Date,
+  HalfFloat,
```
[the remark this answers is not in the document]
Comment (zzt93) [reply src/main/java/org/springframework/data/elasticsearch/annotations/FieldType.java]: Fixed

File: src/main/java/org/springframework/data/elasticsearch/annotations/FieldType.java [hunk 3]
```diff
@@ -19,12 +19,17 @@
...
  */
 public enum FieldType {
 	Text,
+  Byte, 
+  Short, 
 	Integer,
 	Long,
 	Date,
+  HalfFloat, 
+  ScaledFloat,
```
[the remark this answers is not in the document]
Comment (zzt93) [reply src/main/java/org/springframework/data/elasticsearch/annotations/FieldType.java]: Fixed

**Comment by sothawo** [comment]
I just wrote a small test before merging this and noticed that a field with type _scaled_float_ needs to have an attribute _scaling_factor_.  I think it would be more appropriate to have an annotation`@ScaledFloat(scaling_factor = XXX)` and adapt the `MappingBuilder` to process this annotation.
You can either implement it in this ticket, or remove the addition of _scaled_float_ here and I put that in a new ticket.

**Comment by zzt93** [comment]
@sothawo Can you share your test for I to check how to add a scaling factor for it?

**Comment by sothawo** [comment]
Basically I would prefer to have a custom annotation which just has the _scaling_factor_ as parameter. But you are right, the `@Field` annotation already has some custom attributes that are applied when the type is `FieldType.Date`, so to be consequent that would mean, that we had to introduce a custom `@DateField` annotation as well, and that would be a breaking API change.

So I think it is ok to add the _scaling_factor_ attribute to the `@Field` annotation - Java Type Integer - , and in the `MappingBuilder.addFieldMappingParameters` method process this additional attribute in case of a _scaled_float_ field (must not be null, greater than or equal 1).

In the `MappingBuilderTests` class I'd add the field to the `AbstractInheritedEntity` class and then check the mapping in the `shouldBuildMappingWithSuperclass` method.

About the tests I did: I changed the `FieldType`class according to your PR (was only 4 lines) and then added fields with the new types to the `Book` class of the `MappingBuilderTest`. That broke some test because of the mising _scaling_factor_ attribute.

**Pull request merged**